discord-bot/deploy.sh

```diff
@@ -0,0 +1,43 @@
+#!/bin/bash
+
+SERVICE_NAME="manifold-discord-bot"
+
+ENV=${1:-dev}
+case $ENV in
+    dev)
+      ENVIRONMENT=DEV
+      GCLOUD_CPU=1
+      GCLOUD_PROJECT=dev-mantic-markets ;;
+    prod)
+      ENVIRONMENT=PROD
+      GCLOUD_CPU=4
+      GCLOUD_PROJECT=mantic-markets ;;
+    *)
+      echo "Invalid environment; must be dev or prod."
+      exit 1
+esac
+
+# note that we deploy this onto us-east-4, nearest to supabase, not us-central like most stuff
+
+yarn build && \
+  gcloud builds submit . \
+         --tag gcr.io/${GCLOUD_PROJECT}/${SERVICE_NAME} \
+         --project ${GCLOUD_PROJECT} && \
+  gcloud beta run deploy ${SERVICE_NAME} \
+         --image gcr.io/${GCLOUD_PROJECT}/${SERVICE_NAME} \
+         --project ${GCLOUD_PROJECT} \
+         --region us-east4 \
+         --set-env-vars ENVIRONMENT=${ENVIRONMENT} \
+         --set-env-vars GOOGLE_CLOUD_PROJECT=${GCLOUD_PROJECT} \
+         --set-secrets SUPABASE_PASSWORD=SUPABASE_PASSWORD:latest \
+         --execution-environment gen2 \
+         --cpu ${GCLOUD_CPU} \
+         --memory 2Gi \
+         --concurrency 1000 \
+         --min-instances 1 \
+         --no-allow-unauthenticated \
+         --no-cpu-throttling
```

Comment:
`no-cpu-throttling` is the setting for "allocate CPU even when not serving HTTP requests" so I think we are all set on that score.